packages/node_modules/pouchdb-adapter-indexeddb/src/find.js

```diff
@@ -0,0 +1,256 @@
+'use strict';
+
+import {
+  DOC_STORE,
+  idbError,
+  rawIndexFields,
+  naturalIndexName
+} from './util';
+
+import {
+  IDB_NULL,
+  IDB_TRUE,
+  IDB_FALSE,
+} from './rewrite';
+
+// Adapted from
+// https://github.com/pouchdb/pouchdb/blob/master/packages/node_modules/pouchdb-find/src/adapters/local/find/query-planner.js#L20-L24
+// This could change / improve in the future?
+var COUCH_COLLATE_LO = null;
+var COUCH_COLLATE_HI = '\uffff'; // actually used as {"\uffff": {}}
+
+// Adapted from: https://www.w3.org/TR/IndexedDB/#compare-two-keys
+// Importantly, *there is no upper bound possible* in idb. The ideal data
+// structure an infintely deep array:
+//   var IDB_COLLATE_HI = []; IDB_COLLATE_HI.push(IDB_COLLATE_HI)
+// But IDBKeyRange is not a fan of shenanigans, so I've just gone with 12 layers
+// because it looks nice and surely that's enough!
+var IDB_COLLATE_LO = Number.NEGATIVE_INFINITY;
+var IDB_COLLATE_HI = [[[[[[[[[[[[]]]]]]]]]]]];
+
+//
+// TODO: this should be made offical somewhere and used by AllDocs / get /
+// changes etc as well.
+//
+function externaliseRecord(idbDoc) {
```

Comment:
Do you still want to make this official and make it the only function we use to externalise?

Reply:
It's on my list of TODOs to look into at some point yeah. It would just be a minor cleanup, but it would also help to double-check, document and standardise that we shouldn't use `doc.data` when returning docs to callers.